src/main/java/io/vertx/ext/stomp/impl/DefaultStompHandler.java

```diff
@@ -0,0 +1,563 @@
+    return transaction != null && transactions.remove(transaction);
+  }
+
+  @Override
+  public synchronized StompServerHandler unregisterTransactionsFromConnection(StompServerConnection connection) {
+    transactions.stream()
+        .filter(transaction -> transaction.connection().equals(connection))
+        .sorted() // Avoid using baking up collection. TODO Test we dont have concurrent modification exception.
+        .forEach(transactions::remove);
+    return this;
+  }
+
+  @Override
+  public synchronized List<Transaction> getTransactions() {
+    return transactions;
+  }
+
+  @Override
+  public synchronized Subscription getSubscription(StompServerConnection connection, String ackId) {
+    return subscriptions.values().stream().flatMap(List::stream).filter(subscription ->
+        subscription.connection().equals(connection) && subscription.contains(ackId)).findFirst().orElse(null);
+  }
+
+  @Override
+  public StompServerHandler onAck(Subscription subscription, List<Frame> messages) {
+    AcknowledgmentHandler handler;
+    synchronized (this) {
+      handler = onAckHandler;
+    }
+    if (handler != null) {
+      handler.handle(subscription, messages);
+    }
+    return this;
+  }
+
+  @Override
+  public StompServerHandler onNack(Subscription subscription, List<Frame> messages) {
+    AcknowledgmentHandler handler;
+    synchronized (this) {
+      handler = onNackHandler;
+    }
+    if (handler != null) {
+      handler.handle(subscription, messages);
+    }
+    return this;
+  }
+
+  @Override
+  public synchronized StompServerHandler onAckHandler(AcknowledgmentHandler handler) {
+    this.onAckHandler = handler;
+    return this;
+  }
+
+  @Override
+  public synchronized StompServerHandler onNackHandler(AcknowledgmentHandler handler) {
+    this.onNackHandler = handler;
+    return this;
+  }
+
+  /**
+   * Allows customizing the action to do when the server needs to send a `PING` to the client. By default it send a
+   * frame containing {@code EOL} (specification). However, you can customize this and send another frame. However,
+   * be aware that this may requires a custom client.
+   * <p/>
+   * The handler will only be called if the connection supports heartbeats.
+   *
+   * @param handler the action to execute when a `PING` needs to be sent.
+   * @return the current {@link StompServerHandler}
+   */
+  @Override
+  public synchronized StompServerHandler pingHandler(Handler<StompServerConnection> handler) {
+    this.pingHandler = handler;
+    return this;
+  }
+
+
+  private synchronized void addSubscription(String destination, Subscription subscription) {
+    List<Subscription> list = subscriptions.get(destination);
+    if (list == null) {
+      list = new ArrayList<>();
+      subscriptions.put(destination, list);
+    }
+    list.add(subscription);
+  }
+
+  private synchronized boolean removeSubscription(String id, StompServerConnection connection) {
```

Comment:
If the connection is closed or just disappears, do the subscriptions for that connection get automatically removed? If not, there is a memory leak.